### DISCUSSION TITLE
Does Apache Cloudstack support MariaDB ?

### DISCUSSION THREAD

**Discussion by ar-andy-01** [apache/cloudstack#9742] [Q&A] [answered]
I want to use MariaDB for my cloudstack apache on my RHEL8 instance.
But when I try to install I see hard requirement of MySQL database( Also the same can be seen in package spec file)
But when I see release note it mention features for MariaDB and old documentation also mentions installation steps with MariaDB (which don't work now)
Thus I am curious is it supported or not ?

**Comment by weizhouapache** [comment]
@ar-andy-01 
yes, mariadb is supported since many years ago.

**Reply by ar-andy-01** [reply]
Thanks for quick reply :)
But installation of clodstack apache management server has now added dependency of mysql
"https://github.com/apache/cloudstack/blob/main/packaging/el8/cloud.spec" [Line 71]
Which does not work with package installer provisioned by cloudstack apache.

**Reply by weizhouapache** [reply]
I think it means mysql client.
maybe it should be `(mysql-client or mariadb-client)`, need double-check

**Reply by ar-andy-01** [reply]
yes, that is for client connection so it should be (mysql or mariadb)

**Reply by ar-andy-01** [reply]
Will this be fixed in next version ?

**Reply by weizhouapache** [reply, marked as answer]
> Will this be fixed in next version ?

@ar-andy-01 
the code change is simple #9941 .
since it requires some testing, I will not be in next 4.20.0.
maybe in 4.19.2 or 4.20.1

**Reply by nlindblo** [reply]
I was just in the process of migrating my existing mysql database to MariaDB when I read this question, can someone please confirm that the constrain above is only for the initial database creation ?

**Reply by weizhouapache** [reply]
@nlindblo 
as I understand,the issue only impacts the users who want to install mariadb-server on the same server for cloudstack-management.
If mariadb-server is installed on another server (I think most environments are set up like it), it should work fine.

**Reply by DaanHoogland** [reply]
#9941 is merged and should address this
